src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,100 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final String INDEX_EXCEPTION = "Index is invalid";
+    private static final String VALUE_EXCEPTION = "Is no such element present";
+    private T[] array;
+    private int size = 0;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public void grow() {
+        T[] growArray = (T[]) new Object[(int) (array.length * 1.5)];
+        System.arraycopy(array, 0, growArray, 0, array.length);
+        array = growArray;
+    }
+
+    public void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException(INDEX_EXCEPTION);
+        }
+    }
+
+    public int checkValue(T value) {
```

Comment:
Why this method called `checkValue` if it returns some array index? And I see that you use it only in `remove(T element)` method, perhaps this logic should not be moved to a separate method?